Swiftz/OptionalExt.swift

```diff
@@ -49,13 +49,13 @@ public func coalesce<T>(all : @autoclosure () -> T? ...) -> T? {
 
 /// Given one or more Optional values, returns the first Optional value that is not nil
 /// and satisfies the approve function when evaulated from left to right
-public func coalesce<T>(approve : T -> Bool) -> (@autoclosure () -> T? ...) -> T? {
-	return { all in
+public func coalesce<T>
```

Comment:
I leave that up to you :) yeah there's no way to have implicit laziness without autoclosure...